File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        
        if(root==nullptr)return 0;
        
        if(root->left==nullptr && root->right==nullptr)return 1;
        
        int leftd = INT_MIN;
        if(root->left!=nullptr){
            leftd = maxDepth(root->left)+1;
        }
        if(root->right!=nullptr){
            return max(leftd, maxDepth(root->right)+1);
        }
        return leftd;
        // return max(leftd,rightd);
        
    }
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int maxDepth(TreeNode root) {
        
        if(root==null){
            return 0;
        }
        int ldepth = root.left!=null ? maxDepth(root.left):0;
        int rdepth = root.right!=null ? maxDepth(root.right):0;
        return 1 + Math.max(ldepth, rdepth);
        
    }
}